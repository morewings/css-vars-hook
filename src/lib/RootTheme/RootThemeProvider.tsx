import type {FC, ReactNode} from 'react';
import React, {useMemo} from 'react';

import {ROOT_ID} from '../config';
import type {ThemeType} from '../ThemeType';
import {RootContext} from './RootContext';
import {useRootThemeActions} from './useRootThemeActions';

export type ProviderProps = {
    children: ReactNode;
    theme: ThemeType;
};

export const RootThemeProvider: FC<ProviderProps> = ({children, theme}) => {
    const {setTheme, style, getTheme, getVariable, setVariable, removeVariable} = useRootThemeActions(theme);

    const {Provider} = RootContext;

    const actions = useMemo(
        () => ({setTheme, getTheme, getVariable, setVariable, removeVariable}),
        [setTheme, getTheme, getVariable, setVariable, removeVariable]
    );

    return (
        <Provider value={actions}>
            <div id={ROOT_ID} style={style}>
                {children}
            </div>
        </Provider>
    );
};
