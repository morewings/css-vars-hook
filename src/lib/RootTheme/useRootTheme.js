import {useRootContext} from './RootContext';

export const useRootTheme = () => useRootContext();
