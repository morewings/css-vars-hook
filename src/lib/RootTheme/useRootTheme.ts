import {useRootContext} from './RootContext';

export const useRootTheme = () => {
  return useRootContext();
};
