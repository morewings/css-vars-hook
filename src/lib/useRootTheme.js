import {useInsertionEffect} from 'react';
import {setRootVariable, removeRootVariable} from 'lib/utils';

export const useRootTheme = (theme = {}) => {
  // const root = useRef(document.querySelector(':root'));
  useInsertionEffect(() => {
    Object.keys(theme).forEach(key => {
      setRootVariable(key, theme[key]);
    });
  }, [theme]);
  return {
    setRootVariable,
    removeRootVariable,
  };
};
