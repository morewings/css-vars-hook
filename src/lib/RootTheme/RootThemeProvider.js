import React, {useMemo} from 'react';
import {ROOT_ID} from 'lib/config';
import {RootContext} from './RootContext';
import {useRootThemeActions} from './useRootThemeActions';

const themeDefault = {
  foo: 'bar',
};

export const RootThemeProvider = ({children, theme = themeDefault}) => {
  const {setTheme, style, getTheme, getVariable, setVariable, removeVariable} = useRootThemeActions(theme);

  const {Provider} = RootContext;

  const actions = useMemo(
    () => ({setTheme, style, getTheme, getVariable, setVariable, removeVariable}),
    [setTheme, style, getTheme, getVariable, setVariable, removeVariable]
  );

  return (
    <Provider value={actions}>
      <div id={ROOT_ID} style={style}>
        {children}
      </div>
    </Provider>
  );
};
