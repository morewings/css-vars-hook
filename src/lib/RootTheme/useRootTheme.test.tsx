import React from 'react';
import {renderHook} from '@testing-library/react-hooks';

import {RootThemeProvider} from './RootThemeProvider';
import {useRootTheme} from './useRootTheme';

const theme = {
    foo: 'bar',
};

describe('useRootTheme', () => {
    it.skip('renders', () => {
        const {result} = renderHook(() => useRootTheme(), {
            wrapper: ({children}) => <RootThemeProvider theme={theme}>{children}</RootThemeProvider>,
        });
        expect(result.current).toMatchSnapshot();
        // expect(result.current.getTheme()).toBe(theme);
    });

    it.todo('setTheme');
    it.todo('getTheme');
    it.todo('getVariable');
    it.todo('setVariable');
    it.todo('removeVariable');
});
