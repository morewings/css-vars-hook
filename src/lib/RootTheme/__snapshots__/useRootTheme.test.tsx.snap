// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useRootTheme renders 1`] = `
Object {
  "getTheme": [Function],
  "getVariable": [Function],
  "removeVariable": [Function],
  "setTheme": [Function],
  "setVariable": [Function],
}
`;
